modules/indexer/issues/indexer.go

```diff
@@ -154,70 +83,61 @@ func InitIssueIndexer(syncReindex bool) {
 		pprof.SetGoroutineLabels(ctx)
 		start := time.Now()
 		log.Info("PID %d: Initializing Issue Indexer: %s", os.Getpid(), setting.Indexer.IssueType)
-		var populate bool
+		var (
+			existed bool
+			err     error
+		)
 		switch setting.Indexer.IssueType {
 		case "bleve":
 			defer func() {
 				if err := recover(); err != nil {
 					log.Error("PANIC whilst initializing issue indexer: %v\nStacktrace: %s", err, log.Stack(2))
 					log.Error("The indexer files are likely corrupted and may need to be deleted")
 					log.Error("You can completely remove the %q directory to make Gitea recreate the indexes", setting.Indexer.IssuePath)
-					holder.cancel()
+					globalIndexer = internal.NewDummyIndexer()
 					log.Fatal("PID: %d Unable to initialize the Bleve Issue Indexer at path: %s Error: %v", os.Getpid(), setting.Indexer.IssuePath, err)
 				}
 			}()
-			issueIndexer := NewBleveIndexer(setting.Indexer.IssuePath)
-			exist, err := issueIndexer.Init()
+			issueIndexer := bleve.NewIndexer(setting.Indexer.IssuePath)
+			existed, err = issueIndexer.Init(ctx)
 			if err != nil {
-				holder.cancel()
+				globalIndexer = internal.NewDummyIndexer()
 				log.Fatal("Unable to initialize Bleve Issue Indexer at path: %s Error: %v", setting.Indexer.IssuePath, err)
```

Comment:
What's the difference to
```suggestion
				log.Fatal("Unable to initialize Bleve Issue Indexer at path: %s Error: %v", setting.Indexer.IssuePath, err)
```
?

Reply:
Respect the old code `holder.cancel()` by `globalIndexer = internal.NewDummyIndexer()`. Maybe it's OK to ignore `holder.cancel()`, but I won't do that in this PR.